### PR TITLE
Search: clicking on header search will trigger a redirection to searc…

### PR DESCRIPTION
**Pull request by mbarrenechea**

### CODE REVIEW

File: app/assets/javascripts/views/shared/search.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+(function (App) {
```
Comment (davidsingal) [inline app/assets/javascripts/views/shared/search.js]: We don't need this behavior, right? I think we can remove this file.

File: app/assets/javascripts/pages/static_pages/search.js [hunk 2]
```diff
@@ -1,45 +1,35 @@
...
+    _setEventListeners: function () {
+      this.$searchInput = $('#search-input');
+      this.$searchButton.on('click', this._toggleSearch.bind(this));
+      this.$closeButton.on('click', this._closeSearch.bind(this));
+    },
+
+    _toggleSearch: function () {
+      this.$searchContainer.toggleClass('-open');
+      if (this.$searchContainer.hasClass('-open')) {
+        $('.c-modal-search').bind('transitionend webkitTransitionEnd oTransitionEnd MSTransitionEnd', function(e){
```
Comment (davidsingal) [inline app/assets/javascripts/pages/static_pages/search.js]: why not `on` instead `bind`?

File: app/assets/javascripts/pages/static_pages/search.js [hunk 3]
```diff
@@ -1,45 +1,35 @@
...
+          $('#search-input').focus();
```
Comment (davidsingal) [inline app/assets/javascripts/pages/static_pages/search.js]: you should use `this.$searchInput`, take care of the scope of `this`

File: app/assets/javascripts/pages/static_pages/search.js (hunk 2 above)
Comment (mluena) [reply app/assets/javascripts/pages/static_pages/search.js]: didn't know .bind() pointed to .on() until now . changed!!

**Pull request merged**